### PR TITLE
Prevented SQL injection in product sales

### PR DESCRIPTION
**Pull request by thierrymarianne**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Prevent SQL injections |
| Type? | bug fix |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-1464 |
| How to test? | See ticket |

### CODE REVIEW

File: classes/ProductSale.php [hunk 1]
```diff
@@ -132,6 +132,14 @@ public static function getBestSales($id_lang, $page_number = 0, $nb_products = 1
         }
 
         if ($final_order_by != 'price') {
+            if (!Validate::isOrderBy($order_by)) {
+                $order_by = 'cp.position';
+            }
+
+            if (!Validate::isOrderWay($order_way)) {
+                $order_way = 'ASC';
+            }
```
Comment (aleeks) [inline classes/ProductSale.php]: @thierrymarianne  How did you chose to " position " and " asc " ?
Previously, in the function, there is default $order_by (quantity) & $order_way (desc) if the fields were empty. It's not better to take this ?

```
if (is_null($order_by)) {
    $order_by = 'quantity';
    $order_table = 'ps';
}

if (is_null($order_way) || $order_by == 'sales') {
    $order_way = 'DESC';
}
```

**Review by aleeks — CHANGES_REQUESTED** [review]
Check comment

**Review by aleeks — APPROVED** [review]
Great!

**Comment by mickaelandrieu** [comment]
Thank you @thierrymarianne

**Pull request merged**